API.Microsoft.Extensions.AsyncState/NoDocs/Microsoft.Extensions.AsyncState/IAsyncLocalContext.cs

```diff
@@ -0,0 +1,12 @@
+// Assembly 'Microsoft.Extensions.AsyncState'
+
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Extensions.AsyncState;
+
+[Experimental("EXTEXP0006", UrlFormat = "https://aka.ms/dotnet-extensions-warnings/{0}")]
```

Comment:
Experimental

Reply:
Do we need EdtorBrowsable.Never? This looks like a marker interface; could we use keyd services instead of it?

Reply:
Really, I don't understand why this interface exists, why isn't IAsyncContext sufficient?

@tekian Do you understand this?

Reply:
@andrey-noskov might also know

Reply:
`IAsyncLocalContext` is an interface used to make `AsyncContext` (which is a `IAsyncContext`  flavor that uses `AsyncLocal` to store the state) injectable via DI, so that we can inject in into `AsyncContextHttpContext` to support the case when the context is used in a HTTP application but outside of the context of a specific http request.

Reply:
Approved.
Remove EditorBrowsable.Never